lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -420,55 +434,72 @@ export class UserAgentApplication {
         scopes = [this.clientId];
       }
 
-      const scope = scopes.join(" ").toLowerCase();
-      var popUpWindow = this.openWindow("about:blank", "_blank", 1, this, resolve, reject);
-      if (!popUpWindow) {
-        return;
-      }
-    
-      this._loginInProgress = true;
-
-      this.authorityInstance.ResolveEndpointsAsync().then(() => {
-        const authenticationRequest = new AuthenticationRequestParameters(this.authorityInstance, this.clientId, scopes, ResponseTypes.id_token, this._redirectUri, this._state);
-        if (extraQueryParameters) {
-          authenticationRequest.extraQueryParameters = extraQueryParameters;
+        const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
```

Comment:
Make this a common function

Reply:
+1

Reply:
In loginRedirect, extraQueryParameters is an optional parameter. In the case when we call ATS inside loginRedirect, I see a new extraQueryParam variable which is created with upn and domain_hint added added from adal's id_token and passed to the ATS but the extraQueryParameters passed to the loginRedirect api gets completely ignored. I believe we should append the login_hint and domain_hint params to the user provided extraQueryParameters and have a check if these fields are already present similar to what we do in the else path when we don't call ATS in loginRedirect.

Reply:
@rohitnarula7176 Agree to pass user's extraQueryParameter.